source/segment_context_impl.cc

```diff
@@ -251,6 +252,15 @@ SegmentObject SegmentContextImpl::createSegmentObject() {
   return obj;
 }
 
+bool SegmentContextImpl::readyToSend() {
```

Comment:
I want to confirm about the purpose of this method. In the previous(other languages) implementations, the condition to determine sending segment is
**All spans created in this context(linked to this segment) have stopped**

What is the special case of async for cpp? In Java, we take care of this by using a set of separated APIs, because it used ThreadLocal(binding context with thread), but I believe cpp is not that thing.
Java APIs, https://github.com/apache/skywalking/blob/master/docs/en/guides/Java-Plugin-Development-Guide.md#async-span-apis

Reply:
This had introduced not for async context. Only to achieve the requirement that all spans belong to this segment must be stopped.

Reply:
And, current envoy's implementation has guarantee not to be sent if current segment doesn't have unfinished span. So It was introduced for backward compatibility.

Reply:
Then, what is the real case the span doesn't stop? Is that caused by user's codes?

Reply:
Yes. the problem can be caused from user's codes in the current implementation. (User may forget to stop them) I'd like to introduce RAII (start span and automatically end span when current scope finished) based approach to protect this but not now.